core/src/main/java/org/apache/carbondata/core/datamap/DataMapChooser.java

```diff
@@ -291,8 +295,8 @@ private boolean contains(DataMapMeta mapMeta, List<ColumnExpression> columnExpre
     }
     boolean contains = true;
     for (ColumnExpression expression : columnExpressions) {
-      if (!mapMeta.getIndexedColumns().contains(expression.getColumnName()) || !mapMeta
-          .getOptimizedOperation().containsAll(expressionTypes)) {
+      if (!mapMeta.getIndexedColumnNames().contains(expression.getColumnName()) ||
+          !mapMeta.getOptimizedOperation().containsAll(expressionTypes)) {
```

Comment:
What does `expressionTypes` mean here?
In an expression, we just if the column in ***this*** expression is in indexedColumns as well as the expressionType in ***this*** exception. But it seems that the `expressionTypes` here does not belong to ***this*** expression.

Reply:
expressionTypes is from the FilterResolverIntf. So it is from the expression of the user query. See selectDataMap function